docs/diagrams/CleanSequenceDiagram.puml

```diff
@@ -0,0 +1,71 @@
+@startuml
+!include style.puml
+skinparam ArrowFontStyle plain
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant "c:CleanCommand" as CleanCommand LOGIC_COLOR
+participant "p:GradYearPredicate" as GradYearPredicate LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+[-> LogicManager : execute(clean)
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand(clean)
+activate AddressBookParser
+
+create CleanCommand
+AddressBookParser -> CleanCommand
+activate CleanCommand
+
+CleanCommand --> AddressBookParser
+deactivate CleanCommand
+
+AddressBookParser --> LogicManager : c
+deactivate AddressBookParser
+
+LogicManager -> CleanCommand : execute()
+activate CleanCommand
+
+create GradYearPredicate
+CleanCommand -> GradYearPredicate
+activate GradYearPredicate
+
+GradYearPredicate --> CleanCommand : p
+deactivate GradYearPredicate
+
+CleanCommand -> Model : updateFilteredList(p)
+activate Model
+Model --> CleanCommand
+deactivate Model
+
+
+CleanCommand -> Model : getFilteredList()
+activate Model
+Model --> CleanCommand
+deactivate Model
+
+loop until no more persons in filtered list
+CleanCommand -> Model : deletePerson(person)
+activate Model
+Model --> CleanCommand
+deactivate Model
+end loop
+
+CleanCommand -> Model : updateFilteredList()
```

Comment:
I am unsure about this section, we can't include a parameter for this updateFilteredList() unless we get the original predicate as well?